cli/scorecard/score.go

```diff
@@ -148,19 +150,67 @@ func (inventory *InventoryConfig) Score(config *ScoringConfig) error {
 	err = config.attachViolations(auditResult)
 
 	if len(auditResult.Violations) > 0 {
-		fmt.Printf("\n\n%v total issues found\n", len(auditResult.Violations))
-		for _, category := range config.categories {
-			fmt.Printf("\n\n%v: %v issues found\n", category.Name, category.Count())
-			fmt.Printf("----------\n")
-			for _, cv := range category.constraints {
-				fmt.Printf("%v: %v issues\n", cv.GetName(), cv.Count())
-				for _, v := range cv.Violations {
-					fmt.Printf("- %v\n\n",
-						v.Message,
-					)
-					Log.Debug("Violation metadata", "metadata", v.GetMetadata())
+		content := ""
+		switch outputFormat {
+		case "json":
+			type violationOutput struct {
+				Category   string
+				Constraint string
+				Resource   string
+				Message    string
+			}
+			var vOutput violationOutput
+			for _, category := range config.categories {
+				for _, cv := range category.constraints {
+					for _, v := range cv.Violations {
+						vOutput.Category = category.Name
+						vOutput.Constraint = v.Constraint
+						vOutput.Resource = v.Resource
+						vOutput.Message = v.Message
+						byteContent, err := json.MarshalIndent(vOutput, "", "  ")
+						if err != nil {
+							return err
+						}
+						content += string(byteContent) + "\n"
+						Log.Debug("Violation metadata", "metadata", v.GetMetadata())
+					}
+				}
+			}
+		case "csv":
+			content = "Category,Constraint,Resource,Message\n"
+			for _, category := range config.categories {
+				for _, cv := range category.constraints {
+					for _, v := range cv.Violations {
+						content += fmt.Sprintf("%v,%v,%v,%v\n", category.Name, v.Constraint, v.Resource, v.Message)
```

Comment:
I think we should use Golang's CSV writer instead of trying to manually construct ourselves.

We could define a `ToCSV` function on the `RichViolation` type which can create a slice for easy output. https://stackoverflow.com/questions/43301956/convert-struct-to-slice-of-strings

Reply:
used csv writer in 18e499c
but RichViolation is not used for csv output because metadata is in the struct, I'd prefer csv to include simply string fields only and it's easier to just call those fields.